src/parse/rules/misc_rules.build_defs

```diff
@@ -537,15 +540,27 @@ def tarball(name, srcs, out=None, deps=None, subdir=None,
                              'explicit flag for "compression"')
     else:
         compression, extension = _COMPRESSION.get(compression, ('-a', ''))
+
+    out = out or name + '.tar' + extension
+    # Unlike the rest of the compressions, gzip preserves the file's (the tar)
+    # timestamp by default, making the output unreproducible.
+    # Hence the compression is done with the -n flag in a separate step
+    if extension == '.gz' or out.endswith('.gz'):
+        archive_cmd = 'tar -c * | gzip -n > $OUT'
+    else:
+        archive_cmd = 'tar %s -cf $OUT *' % compression
+
     build_rule(
         name=name,
         cmd=' && '.join([
             'mkdir -p _tmp/' + subdir,
-            'cp -r $(locations :_%s#files) _tmp/%s' % (name, subdir),
+            'cp -pr $(locations :_%s#files) _tmp/%s' % (name, subdir),
```

Comment:
Hm, the cache having different mode bits does not sound right; I'll keep an eye on it next week. It shouldn't break, though, because `cp` preserves them by default, the above change just made it explicit. 
In general, I think the mode bits are useful to preserve for executables and probably something that should affect the output hash.